swagger_to/style.py

```diff
@@ -0,0 +1,224 @@
+            errs.extend(_check_recursively_cases(typedef=prop.typedef, visited=visited))
+
+    return errs
+
+
+def _check_recursively_descriptions(typedef: swagger_to.intermediate.Typedef,
+                                    visited: MutableSet[swagger_to.intermediate.Typedef]) -> List[str]:
+    """
+    Checks the typedef's adherence to the description conventions.
+
+    :param typedef: to be translated
+    :param visited: already seen typedefs
+    :return: the list of failed checks
+    """
+    errs = []  # type: List[str]
+    if typedef in visited:
+        return errs
+
+    visited.add(typedef)
+
+    if isinstance(typedef, swagger_to.intermediate.Primitivedef):
+        pass
+
+    elif isinstance(typedef, swagger_to.intermediate.Arraydef):
+        errs.extend(_check_recursively_descriptions(typedef=typedef.items, visited=visited))
+
+    elif isinstance(typedef, swagger_to.intermediate.Mapdef):
+        errs.extend(_check_recursively_descriptions(typedef=typedef.values, visited=visited))
+
+    elif isinstance(typedef, swagger_to.intermediate.Objectdef):
+        if not check_description(typedef.description):
+            errs.append("Description \"{}\" must begin with a lower case and end with a period.".format(
+                typedef.description))
+
+        for prop in typedef.properties.values():
+            if not check_description(prop.description):
+                errs.append("Description \"{}\" must begin with a lower case and end with a period.".format(
+                    prop.description))
+                errs.extend(_check_recursively_descriptions(typedef=prop.typedef, visited=visited))
+
+    return errs
+
+
+def check_endpoint_path_responses(endpoints: List[swagger_to.intermediate.Endpoint]) -> List[str]:
+    """
+    Checks whether the endpoints conform to the conventions for paths and responses.
+
+    :param endpoints: swagger endpoint definitions
+    :return: the list of failed checks
+    """
+    errs = []  # type: List[str]
+
+    for endpoint in endpoints:
+        if not endpoint.path.startswith("/"):
+            errs.append("Path {} must begin with a slash.".format(endpoint.path))
+        if "200" not in endpoint.responses.keys():
+            errs.append("Path {} must include response 200.".format(endpoint.path))
+        if "default" not in endpoint.responses.keys():
+            errs.append("Path {} must include default response.".format(endpoint.path))
+
+    return errs
+
+
+def check_description(description: str) -> bool:
+    """
+    Checks whether a description is well-styled.
+
+    :param description: the description
+    :return: true if the description's first alphanumeric letter is lowercase and the description ends with a period.
+    """
+    if description == "":
+        return True
+
+    trimmed = description.replace("|", "").strip()
+
+    return trimmed[:1].islower() and trimmed.endswith(".")
```

Comment:
I'd follow pydocstyle here. Here are the extra checks I'd add:
* There should be no trailing and heading new lines in the descriptions.
* There should be no trailing white spaces and tabs.
* The first word needs to be in present tense (ends with "s"). 
* The first line needs to end with a dot. 
* If there are multiple lines, there needs to be a new line between the first and the second paragraph in order to enforce that the first sentence is a summary and the second paragraph is a description. This is equal to first line non-empty, second line empty, third line non-empty.